tilequeue/query/common.py

```diff
@@ -20,6 +21,49 @@ def allows_shape_type(self, shape):
         return typ in self.shape_types
 
 
+class ShapeType(Enum):
+    point = 1
+    line = 2
+    polygon = 3
+
+
+# determine the shape type from the raw WKB bytes. this means we don't have to
+# parse the WKB, which can be an expensive operation for large polygons.
+def wkb_shape_type(wkb):
+    reverse = ord(wkb[0]) == 1
+    type_bytes = map(ord, wkb[1:5])
+    if reverse:
+        type_bytes.reverse()
+    typ = type_bytes[3]
+    if typ == 1 or typ == 4:
+        return ShapeType.point
+    elif typ == 2 or typ == 5:
+        return ShapeType.line
+    elif typ == 3 or typ == 6:
+        return ShapeType.polygon
+    else:
+        assert False, "WKB shape type %d not understood." % (typ,)
+
+
+def parse_shape_types(inputs):
+    lookup = {
```

Comment:
Might be better to have this `lookup` stored somewhere outside the function so it doesn't have to get re-created for every call of `parse_shape_types()`. It might fit in as a `@classmethod` on `ShapeType`?

Reply:
We might be able to be even more clever by leaning on enums.

```python
  class ShapeType(Enum):
      point = 1
      line = 2
      polygon = 3
  
      multipoint = 1
      linestring = 2
      multilinestring = 2
      polygon = 3
      multipolygon = 3
```

Lookup is then just:
```python
ShapeType[value.lower()]
```
which raises a `KeyError` if not found.

Although the code as it is should work, I don't think this is a good idea since it can lead to some confusing behavior:
```python
>>> ShapeType.multipolygon
<ShapeType.polygon: 3>
```

Anyway, just mentioning this but I like @iandees more. We could even just stick the lookup at global scope if the interaction with sticking it on enums is awkward.